httpx/_content_streams.py

```diff
@@ -207,17 +206,34 @@ def __init__(self, name: str, value: typing.Union[str, bytes]) -> None:
             self.name = name
             self.value = value
 
+        @functools.lru_cache(1)
         def render_headers(self) -> bytes:
             name = format_form_param("name", self.name)
             return b"".join([b"Content-Disposition: form-data; ", name, b"\r\n\r\n"])
 
+        @functools.lru_cache(1)
```

Comment:
These functions are simple enough that I'd expect a `@functools.lru_cache(1)` is likely to introduce more overhead than it'll save. We *might* consider either:

* Dropping the `@functools.lru_cache(1)` on trivial method implementations.
* Dropping the `@functools.lru_cache` cases everywhere for this PR, focusing strictly on correctness, and *potentially* adding them in as an incremental improvement sperately. (It's feasible that there's stateful approaches that we might consider instead of `lru_cache` wrapping.)

Any thoughts?

Reply:
I actually initially used private unset attributes + `if not hasattr(self, "_headers")`, but decided the `lru_cache` way was a bit cleaner while functionally identical.

I guess we can revert to that correct-first approach though, sure.